src/EbmlSInteger.cpp

```diff
@@ -116,7 +116,7 @@ uint64 EbmlSInteger::UpdateSize(bool bWithDefault, bool /* bForceRender */)
     SetSize_(2);
   } else if (Value <= 0x7FFFFF && Value >= (-0x800000)) {
     SetSize_(3);
-  } else if (Value <= EBML_PRETTYLONGINT(0x7FFFFFFF) && Value >= (EBML_PRETTYLONGINT(-0x80000000))) {
+  } else if (Value <= EBML_PRETTYLONGINT(0x7FFFFFFF) && Value >= (EBML_PRETTYLONGINT(-(int64)0x80000000))) {
```

Comment:
This looks like the wrong kind of fix to me. The `EBML_PRETTYLONGINT` macro is supposed to hide compiler-specific requirements to numbers. The correct fix might just be to add the correct suffix (`ll`) to the number; that's what `EBML_PRETTYLONGINT` does, but only for GCC at the moment.

So please look into extending the [definition of `EBML_PRETTYLONGINT`](https://github.com/Matroska-Org/libebml/blob/master/ebml/EbmlConfig.h#L94) to MSVC, too. Thanks.